From a860dd7e51ab799d3f6be9963bc85ed4448db86b Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 22 Feb 2024 14:10:35 +0100
Subject: [PATCH 0935/1141] dt-bindings: display: update dt-bindings for STM32
 DSI device

Add "st,stm32mp25-dsi" compatible new power supplies.

Change-Id: I69953f69396eeacb3b3896046e6b107fa2209798
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/362682
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../bindings/display/st,stm32-dsi.yaml        | 42 +++++++++++++++----
 1 file changed, 34 insertions(+), 8 deletions(-)

--- a/Documentation/devicetree/bindings/display/st,stm32-dsi.yaml
+++ b/Documentation/devicetree/bindings/display/st,stm32-dsi.yaml
@@ -13,12 +13,11 @@ maintainers:
 description:
   The STMicroelectronics STM32 DSI controller uses the Synopsys DesignWare MIPI-DSI host controller.
 
-allOf:
-  - $ref: dsi-controller.yaml#
-
 properties:
   compatible:
-    const: st,stm32-dsi
+    enum:
+      - st,stm32-dsi
+      - st,stm32mp25-dsi
 
   reg:
     maxItems: 1
@@ -44,10 +43,6 @@ properties:
     items:
       - const: apb
 
-  phy-dsi-supply:
-    description:
-      Phandle of the regulator that provides the supply voltage.
-
   ports:
     $ref: /schemas/graph.yaml#/properties/ports
 
@@ -73,6 +68,37 @@ properties:
                   - const: 1
                   - const: 2
 
+allOf:
+  - $ref: dsi-controller.yaml#
+
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - st,stm32-dsi
+    then:
+      properties:
+        phy-dsi-supply:
+          description:
+            Phandle of the regulator that provides the supply voltage.
+
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - st,stm32mp25-dsi
+    then:
+      properties:
+        vdd-supply:
+          description:
+            Phandle of the regulator that provides the supply voltage vdd.
+
+        vdda18-supply:
+          description:
+            Phandle of the regulator that provides the supply voltage vdda18.
+
 required:
   - "#address-cells"
   - "#size-cells"
